vllm/entrypoints/openai/serving_chat.py

```diff
@@ -381,9 +395,22 @@ async def chat_completion_full_generator(
             else:
                 logprobs = None
 
+            if request.tool_choice and type(
+                    request.tool_choice) is ChatCompletionNamedToolChoiceParam:
+                message = ChatMessage(
+                    role=role,
+                    content="",
+                    tool_calls=[
+                        ToolCall(function=FunctionCall(
+                            name=request.tool_choice.function.name,
+                            arguments=output.text))
```

Comment:
Question: how does this part work? It looks like it's just taking the model's output and passing that in as the arguments, but how are we guarantee that the output matches the argument's requirements?

Reply:
Outlines or lm-format-enforcer guarantee that for us.